jenkins/scripts/bare_metal_lab/tasks/cleanup-tasks.yaml

```diff
@@ -0,0 +1,128 @@
+- name: Check required env vars are set
```

Comment:
I am not sure if I do agree with the name of this file vs what it does. I think we could still move back some parts of it to original deploy-lab.yaml and leave it there and leave here only the parts that are really doing the "cleaning up" tasks.

Reply:
We cannot move them to deploy lab, because sequence is important here. What we can do create another playbook that includes tasks which are tagged as "deploy".

Reply:
We clone metal3-dev-env to two different directories depending what we are testing. we need to run make clean from that directory as an essential part of cleanup. So, we cannot move these tasks to deploy-lab.yaml. But during cleanup after successful run, we need to somehow skip cloning and merging metal3-dev-env tasks again.

Reply:
Other two tasks related finding and deleting log files also part of cleanup but we do not need to clean them after successful run. We might need the logs later for debuging or some other purposes. These tasks are not part of deploy tasks because of what they do.